metrics.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 RedHat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// StartMetrics starts the server with registered metrics on the specified endpoint.
+func StartMetrics(config metricsConfig) {
+	// Register Metrics only when the metric List is provided by the operator
+	if config.collectorList != nil {
+		RegisterMetrics(config.collectorList)
+
+		//Execute the user-defined function on how to collect metrics
+		config.recordMetricsFunction()
+	}
+
+	http.Handle(config.metricsPath, prometheus.Handler())
+	metricsPort := ":" + (config.metricsPort)
+	log.Info("Port: %v", metricsPort)
```

Comment:
This will print `Port: :####` which looks messy, we should move this one line up to print `Port: ####`.